dev/com.ibm.ws.security.javaeesec/src/com/ibm/ws/security/javaeesec/properties/ModulePropertiesUtils.java

```diff
@@ -128,9 +128,17 @@ private HttpAuthenticationMechanism getHttpAuthenticationMechanism(boolean logEr
                         if (mppi != null && !mppi.isUnsatisfied() && !mppi.isAmbiguous()) {
                             List<Class> implClassList = mppi.get().getAuthMechClassList();
                             if (implClassList != null) {
-                                if (implClassList.size() == 1) {
-                                    hamClass = implClassList.get(0);
-                                    Bean<HttpAuthenticationMechanism> bean = getBean(beanManager, hamClass);
+                                if (implClassList.size() == 0) {
+                                    Tr.error(tc, "JAVAEESEC_ERROR_NO_HAM", getJ2EEModuleName(), getJ2EEApplicationName());
```

Comment:
This getHttpAuthenticationMechanism method is indirectly called in the path of the BridgeBuilderImpl's buildBridgeIfNeeded method when it checks for,

if (!getModulePropertiesUtils().isHttpAuthenticationMechanism())

So, there should not be error when there are no mechanisms for an app and that is indicated by the "logError" parameter. Therefore, the Tr.error above could be guarded, like this,

```
if (logError) {
    Tr.error(tc, "JAVAEESEC_ERROR_NO_HAM", getJ2EEModuleName(), getJ2EEApplicationName());
}
```